gtsfm/utils/io.py

```diff
@@ -85,3 +84,80 @@ def save_json_file(
         json.dump(data, f, indent=4)
 
 
+def read_bal(file_path: str) -> GtsfmData:
+    """Read a BAL file.
+
+    Args:
+        file_name: file path of the BAL file.
+
+    Returns:
+        The data as an GtsfmData object.
+    """
+    sfm_data = gtsam.readBal(file_path)
+
+    num_images = sfm_data.number_cameras()
+
+    gtsfm_data = GtsfmData(num_images)
+    for i in range(num_images):
+        camera = sfm_data.camera(i)
+        gtsfm_data.add_camera(camera, i)
+    for j in range(sfm_data.number_tracks()):
+        gtsfm_data.add_track(sfm_data.track(j))
+
+    return gtsfm_data
+
+
+def write_cameras(gtsfm_data: GtsfmData, file_path: str) -> None:
```

Comment:
Thanks for the updates, @ayushbaid.

I think it would make more sense to pass the `save_dir` here, since by design the filename must be `cameras.txt` to match COLMAP

Reply:
(and not pass a file_path)